src/platform/packages/shared/response-ops/alerts-table/hooks/use_alerts_table_configuration.tsx

```diff
@@ -0,0 +1,116 @@
+}
+
+/**
+ * Manages the alerts table configuration persistence.
+ * Loads, validates, and saves table settings to storage, returning a `React.useState`-compatible tuple.
+ * Handles invalid configurations by resetting and notifying the user.
+ */
+export const useAlertsTableConfiguration = ({
+  id,
+  configurationStorage,
+  notifications,
+}: UseAlertsTableConfigurationParams) => {
+  const { toasts } = notifications;
+
+  const loadConfiguration = useCallback(() => {
+    if (!configurationStorage) {
+      return null;
+    }
+    const savedConfig = configurationStorage.get(id);
+
+    if (!savedConfig) {
+      return null;
+    }
+
+    try {
+      return alertsTableConfigurationSchema.parse(JSON.parse(savedConfig));
+    } catch (e) {
+      configurationStorage.remove(id);
+      toasts.addWarning({
+        title: i18n.translate('xpack.responseOpsAlertsTable.saveConfigurationError', {
+          defaultMessage: 'Unable to load alerts table configuration',
+        }),
+        text: i18n.translate('xpack.responseOpsAlertsTable.saveConfigurationError', {
+          defaultMessage:
+            'The saved configuration was invalid and has been reset. Any new table customizations will be saved from now on.',
+        }),
+      });
+    }
+
+    return null;
+  }, [configurationStorage, id, toasts]);
+
+  const [configuration, setConfiguration] = useState(loadConfiguration);
+
+  const safeSetConfiguration = useCallback<
+    Dispatch<SetStateAction<AlertsTableConfiguration | null>>
+  >(
+    (setStateAction) => {
+      if (!configurationStorage) {
+        return;
+      }
+      // It's important to use the functional form of setState to reference the old state instead
+      // of using the `configuration` variable from the outer scope, as that would make this callback
+      // referentially unstable and cause an unnecessary configuration save
+      setConfiguration((prevConfig) => {
+        const newConfig =
+          typeof setStateAction === 'function' ? setStateAction(prevConfig) : setStateAction;
+
+        try {
+          // Parsing the configuration ensures that it is valid and contains only the minimal
+          // user overrides
+          const parsedConfig =
+            newConfig === null ? null : alertsTableConfigurationSchema.parse(newConfig);
+          const serializedConfig = JSON.stringify(parsedConfig);
+
+          if (serializedConfig === JSON.stringify(prevConfig)) {
+            // Avoid causing rerenders and saving to storage if the previous and new configurations
+            // are structurally equivalent
+            return prevConfig;
+          }
+          if (parsedConfig === null) {
+            // If the new config is null, remove the saved configuration
+            configurationStorage.remove(id);
+          } else {
+            configurationStorage.set(id, serializedConfig);
+          }
+          return parsedConfig;
+        } catch {
+          // TODO add toast?
```

Comment:
I'm not sure if we can ever fall in this catch tbh, since this is the config coming from the table itself